src/vcpkg/registries.cpp

```diff
@@ -868,50 +861,47 @@ namespace
         return Path(prefix) / port_name.to_string() + ".json";
     }
 
-    ExpectedS<std::vector<VersionDbEntry>> load_versions_file(const Filesystem& fs,
+    ExpectedL<std::vector<VersionDbEntry>> load_versions_file(const Filesystem& fs,
                                                               VersionDbType type,
                                                               const Path& registry_versions,
                                                               StringView port_name,
                                                               const Path& registry_root)
     {
         if (type == VersionDbType::Filesystem && registry_root.empty())
         {
-            Debug::print("Bug in vcpkg; type should never = Filesystem when registry_root is empty.");
-            Checks::exit_fail(VCPKG_LINE_INFO);
+            Checks::unreachable(VCPKG_LINE_INFO, "type should never = Filesystem when registry_root is empty.");
         }
 
         auto versions_file_path = registry_versions / relative_path_to_versions(port_name);
 
         if (!fs.exists(versions_file_path, IgnoreErrors{}))
         {
-            return Strings::format("Couldn't find the versions database file: %s", versions_file_path);
+            return msg::format_error(msgCouldNotFindVersionDatabaseFile, msg::path = versions_file_path);
         }
 
         std::error_code ec;
         auto contents = fs.read_contents(versions_file_path, ec);
         if (ec)
         {
-            return Strings::format(
-                "Error: Failed to load the versions database file %s: %s", versions_file_path, ec.message());
+            return format_filesystem_call_error(ec, "read_contents", {versions_file_path});
         }
 
         auto maybe_versions_json = Json::parse(contents);
-        if (!maybe_versions_json)
+        auto versions_json = maybe_versions_json.get();
+        if (!versions_json)
         {
-            return Strings::format("Error: failed to parse versions file for `%s`: %s",
-                                   port_name,
-                                   maybe_versions_json.error()->to_string());
+            return msg::format_error(msgFailedToParseJson, msg::path = versions_file_path);
```

Comment:
This no longer emits the detailed parse errors. This is especially important for Filesystem registries where the user might be actively editing them.

Reply:
I think basically all of the "msgFailedToParseJson" ones are errors. Stand by...

Reply:
Example: 
![image](https://user-images.githubusercontent.com/1544943/221096603-aa988ad6-ab5b-4fa2-b181-aed3b19fb42e.png)

Reply:
![image](https://user-images.githubusercontent.com/1544943/221097888-b96453dc-b75b-4b72-9c1d-6c4c34ef21a1.png)

Reply:
Example:

![image](https://user-images.githubusercontent.com/1544943/221098538-c8c53c8a-39e1-4987-be29-10375300b942.png)